File: investing-backend/src/main/java/com/davidmuns/investing/dto/PositionSummaryResponse.java
package com.davidmuns.investing.dto;

public record PositionSummaryResponse(
        String name,
        String symbol,
        String type,
        Double totalQuantity,
        Double averagePrice,
        Double currentPrice,
        Double marketValue,
        Double dailyProfitLoss,
        Double dailyProfitLossPercentage,
        Double netProfitLossPercentage,
        Double netProfitLoss
) {
}

File: investing-frontend/src/app/shared/models/position-summary-response.ts
export interface PositionSummaryResponse {
  id: number;
  name: string;
  symbol: string;
  type: string;
  totalQuantity: number;
  averagePrice: number;
  currentPrice: number;
  marketValue: number;
  dailyProfitLoss: number;
  // dailyProfitLossPercentage: number;
  netProfitLossPercentage: number;
  netProfitLoss: number;
  portfolioId: number;
}
